### ISSUE TITLE
API Call is appending "; charset=utf-8" automatically to Header "Content-Type"

### ISSUE THREAD

**Issue by johnpreynoso** [FlutterFlow/flutterflow-issues#3654]
### Can we access your project?

- [ ] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

In an API Call, FlutterFlow is appending a value to Header "Content-Type".

Even if I explicitly set "Content-Type" to be "application/merge-patch+json", once the API call is made, the header value for "Content-Type" becomes "application/merge-patch+json; charset=utf-8".

![001](https://github.com/user-attachments/assets/f3bb3e19-7d4e-4d37-8adf-0c593a1bc666)
![002](https://github.com/user-attachments/assets/52c6bdda-3b27-4a1a-baaa-f516cf35577c)
![003](https://github.com/user-attachments/assets/3a87fe13-65db-47ce-8704-851bbda6c6e0)
![004](https://github.com/user-attachments/assets/81e01cc8-cc36-40e4-83da-63bf5b81f108)

In the specific case I'm working on, the API call receiver specifically requires the "Content-Type" to be exactly "application/merge-patch+json"; otherwise, the call made will receive status "400: Bad Request".

### Expected Behavior

What I was expecting to happen was for FlutterFlow to only send the "Content-Type" value as seen in the Preview below.
![image](https://github.com/user-attachments/assets/c76b2f5c-5b75-41f1-b928-c1e31ec73988)

### Steps to Reproduce

1. In FlutterFlow, go to API Calls.
2. Press "Add" and "Create API Call".
3. Set API Call Name to: Test API Patch Call
4. Set Method Type to: PATCH
5. Set API URL to: https://postman-echo.com/patch
6. Press "Add Header".
7. Provide the following value for the new header: Content-Type: application/merge-patch+json
8. Press on the "Body" tab.
9. For the body type, select: JSON
10. In the body content, input the following: { "text": "Test text here." }
11. Click on "Response & Test".
12. Under "Preview and Test", select "Headers". The value for "Content-Type" should only be "application/merge-patch+json".
13. Under "Preview and Test", select "Body" to confirm the contents input in step 10.
14. Press "Test API Call".
15. Under "Preview and Test", select "Test Response". Note that the value of "content-type" has now changed to "application/merge-patch+json; charset=utf-8".
![005](https://github.com/user-attachments/assets/c7120694-841c-471d-8b8d-2f30e80e3572)

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

Not applicable; this bug is in API Calls.

### Visual documentation

Refer to "Steps to Reproduce".

![image](https://github.com/user-attachments/assets/a6221d1c-49b4-4097-8d49-68c2489ecc89)

![image](https://github.com/user-attachments/assets/e8aa98a0-e633-43c3-a3cc-f5462847acb5)

![image](https://github.com/user-attachments/assets/68f5e2f3-87ec-40c4-94ce-1aaabbd9c5c8)

![image](https://github.com/user-attachments/assets/90357563-f423-4425-9cfa-e1f2a59d7414)

### Environment

```markdown
- FlutterFlow version: v4.1.83+
- Platform: Windows
- Browser name and version: FlutterFlow Windows development app
- Operating system and version affected: Windows 11
```

### Additional Information

This is currently a showstopper. The project is integrating FlutterFlow with a Cloud ERP. 

In the specific case I'm working on, the API call receiver specifically requires the "Content-Type" to be exactly "application/merge-patch+json"; otherwise, the call made will receive status "400: Bad Request".

**Comment by Alezanello** [comment]
Hello!!<br><br>Just to be sure before avdance in testing this issue, have you enabled the "Decode Response as UTF-8"?

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/02086c85-9a44-4855-9826-cc64363e4571/80505ed6-4cda-4c0d-a05d-cf56a14e652c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8wMjA4NmM4NS05YTQ0LTQ4NTUtOTgyNi1jYzY0MzYzZTQ1NzEvODA1MDVlZDYtNGNkYS00YzBkLWEwNWQtY2Y1NmExNGU2NTJjIiwiaWF0IjoxNzI0MjQ0ODI2LCJleHAiOjMzMjk0ODA0ODI2fQ.Y3t7Urx5e6ItyZdYbMx7xx6uhIy4gE1cu-ejyWZuGyA " alt="image.png" width="662" height="540" />

**Comment by johnpreynoso** [comment]
> Hello!!Just to be sure before avdance in testing this issue, have you enabled the "Decode Response as UTF-8"?
> 
> <img alt="image.png" width="662" height="540" src="https://camo.githubusercontent.com/eaafca227a8a2bfb0815534b5f4d6aaf64f9e25cc573e318cec15e54d834f91f/68747470733a2f2f75706c6f6164732e6c696e6561722e6170702f33323431303839642d373336612d343432322d393030622d3035316537326163636462652f30323038366338352d396134342d343835352d393832362d6363363433363365343537312f38303530356564362d346364612d346330642d613035642d6366353661313465363532633f7369676e61747572653d65794a68624763694f694a49557a49314e694973496e523563434936496b705856434a392e65794a775958526f496a6f694c7a4d794e4445774f446c6b4c54637a4e6d45744e4451794d6930354d4442694c5441314d5755334d6d466a593252695a5338774d6a41344e6d4d344e533035595451304c5451344e5455744f5467794e69316a597a59304d7a597a5a5451314e7a45764f4441314d44566c5a4459744e474e6b59533030597a426b4c5745774e575174593259314e6d45784e4755324e544a6a4969776961574630496a6f784e7a49304d6a51304f4449324c434a6c654841694f6a4d7a4d6a6b304f4441304f44493266512e593374375572783565364974795a6459624d7837787836756849793467453163752d656a79575a7547794120">

Hi, Alezanello.

I do not have that option available since I had to make the call Private.
![image](https://github.com/user-attachments/assets/a8f0b307-99b1-443d-9a1d-37da409e16ee)

**Comment by johnpreynoso** [comment]
Adding a comment to prevent issue closure (this is still open).

**Comment by johnpreynoso** [comment]
This program behavior is still happening in v4.1.85+.

**Comment by johnpreynoso** [comment]
Program behavior still happens in v4.1.88+.
![image](https://github.com/user-attachments/assets/21d09264-5478-477e-b927-705b9578f0cc)

**Comment by Alezanello** [comment]
Hello!

I apologize for the delayed response. I was able to prevent the charset issue in the header by enabling the "Encode Body as UTF-8" option. Could you please give it a try and let me know if it works?

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/37a82d23-7378-46a2-abb5-370f6ddbeac9/486b5a7c-1ad1-4f0b-9c6d-ffe016ee83c9?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8zN2E4MmQyMy03Mzc4LTQ2YTItYWJiNS0zNzBmNmRkYmVhYzkvNDg2YjVhN2MtMWFkMS00ZjBiLTljNmQtZmZlMDE2ZWU4M2M5IiwiaWF0IjoxNzI2NzY2NjI1LCJleHAiOjMzMjk3MzI2NjI1fQ.Y6wja5hHZuuRbRh-7gR0eJhihkogoeX7dCdCUoyjAFE " alt="encodeBody.gif" width="1901" height="912" />

Thank you!

**Comment by johnpreynoso** [comment]
Hi, @Alezanello.

First, I switched off "Make Private" and then turned on "Encode Body as UTF-8 Bytes". The result content type was good.

![image](https://github.com/user-attachments/assets/bbe88e3c-9825-437d-8400-7074cb15cff5)

Next, I went back to Call Definition > Advanced Settings, and also turned on "Make Private". I then had to deploy this (since it has to be built as a cloud function in Google Firebase). Test result was the same.

![image](https://github.com/user-attachments/assets/a5eae773-8f3e-40ee-9279-be6148a308c6)

Issue resolved. Thank you.

The only point of improvement I can suggest here is to have the "Encode Body as UTF-8 Bytes" option visible even when "Make Private" is turned on. I was under the impression that turning on "Make Private" disables the "Encode Body as UTF-8 Bytes" option.

Again, thank you.
